custom_components/evse_load_balancer/chargers/zaptec_charger.py

```diff
@@ -71,15 +72,29 @@ def set_phase_mode(self, mode: PhaseMode, _phase: Phase | None = None) -> None:
         # https://github.com/dirkgroenen/hass-evse-load-balancer/issues/9
 
     async def set_current_limit(self, limit: dict[Phase, int]) -> None:
-        """Set the current limit for the Zaptec charger."""
-        entity_id = self._get_entity_id_by_translation_key(
-            ZaptecEntityMap.ChargingCurrent
+        """Set the current limit for the Zaptec installation."""
+        entity_id = self._get_entity_id_by_translation_key(ZaptecEntityMap.Charger)
+        state = self.hass.states.get(entity_id)
+        if not state or "installation_id" not in state.attributes:
+            _LOGGER.exception(
+                "installation_id not found in attributes for entity %s", entity_id
+            )
+        installation_id = state.attributes["installation_id"]
```

Comment:
I think this can be done a lot simpler and more future-proof:
```suggestion
        installation_id = self.device.via_device_id
```

Reply:
I must admit, I'm not completely sure how your proposal act's out, but I'll try to state my intention: 
According to [Zaptec API guidelines](https://docs.zaptec.com/docs/api-usage-guidelines#8-dynamic-load-balancing), it is reccommended to do the changes at the installation level. This is also covers scenarios where multiple zaptec chargers are installed in the same installation. 
As the integration uses the charger entity mainly, the need of the installation_id only emerges when need of setting the installation max current. 
As far as I knew, the binary_sensor.zaptec_charger was the only place I could find a reference to the actual installation, and by that way successfully be able to set the current limit by the zaptec.limit_current action. 

I would probably concider myself a novice within HA programming, but I made an attempt, and hope it could start the crative process towards a solution. 

I fully understand, and i've also read/noticed the change in policy that you [refer to](https://github.com/custom-components/zaptec/pull/295#issuecomment-3214775023). Maybe there's another way of 'pointing' to the installation device instead?

Reply:
The way the zaptec integration is set up, the installation device is the parent device of one (or more) charger device(s). `self.device` here is (unless I've misread something) the configured charger device. The `via_device` is the parent device (ie the installation), so `via_device_id` should be the installation id you are looking for.

(this is with the caveat that I haven't made any attempt to set up a dev environment for this repo, and I have only been programming HA for ~3 months myself)